storybookjs/storybook
# Fixed React Fiber deprecation of using ref with stateless function components

Issue: #1539

In short, React Fiber deprecated the ability to add ref to a stateless function component.


## What I did

Converted `LeftPanel` and `Shortcuts` from stateless function components to `React.Component`s.
It should be said that it seems the documentations of developing storybook slightly outdated and missing some information. 
Another thing to mention is that I had to use `eslint-disable` for the components since according to some ESLint React plugin rule they shouldn't be a regular `React.Component`.

I hope I did an okay fix. Please, let me know.


## How to test

1. clone the repo with fix
2. run `npm install`
3. run `npm link`
4. go to `lib/ui` and run there `npm link`
5. go to `app/react` and over there do the following:
6. run `npm i react@next react-dom@next --only=dev`
7. run `npm link '@storybook/ui'
8. run `npm link`

After the flowing you should create a React project using React Fiber.
Needs over there to set-up storybook and then run `npm link '@storybook/react'`.
The errors for issue #1539 should gone


Fix issue 1539
